ecp5/pack.cc

```diff
@@ -2706,6 +2721,80 @@ class Ecp5Packer
                     copy_constraint(ci, id_CLK1, id_ECSOUT, 1);
                 } else if (ci->type == id_DCCA) {
                     copy_constraint(ci, id_CLKI, id_CLKO, 1);
+                } else if (ci->type == id_DCSC) {
+                    if ((!ci->ports.count(id_CLK0) && !ci->ports.count(id_CLK1)) || !ci->ports.count(id_DCSOUT))
+                        continue;
+                    auto mode = str_or_default(ci->params, id_DCSMODE, "POS");
+
+                    // TODO: We can do this simplification if we know
+                    // MODESEL is tied to 0. I don't know how I can check this
+                    // if (mode == "CLK0_LOW" || mode == "CLK0_HIGH" || mode == "CLK0") {
```

Comment:
@gatecat 
Technically if `CLKx_LOW` then `low.max_delay` is infinity. SImilar case for `CLKx_HIGH` then `high.max_delay` is infinity. I felt it's not useful to model this here. But I'm happy to proven wrong.

Reply:
Yes, I don't see any case where this would matter.